chromadb/test/property/test_filtering.py

```diff
@@ -63,51 +64,110 @@ def _filter_where_doc_clause(clause, doc):
                 ids.discard(recordset["ids"][i])
 
     return list(ids)
 
 
-collection_st = st.shared(strategies.collections(add_filterable_data=True), key="coll")
-recordset_st = st.shared(strategies.recordsets(collection_st,
-                                                max_size=1000), key="recordset")
-
-
-@settings(suppress_health_check=[HealthCheck.function_scoped_fixture,
-                                 HealthCheck.large_base_example])
-@given(collection=collection_st,
-       recordset=recordset_st,
-       filters=st.lists(strategies.filters(collection_st, recordset_st), min_size=1))
-def test_filterable_metadata(caplog, api, collection, recordset, filters):
+collection_st = st.shared(
+    strategies.collections(add_filterable_data=True, with_hnsw_params=True), key="coll"
+)
+recordset_st = st.shared(
+    strategies.recordsets(collection_st, max_size=1000), key="recordset"
+)
+
+
+@settings(
+    suppress_health_check=[
+        HealthCheck.function_scoped_fixture,
+        HealthCheck.large_base_example,
+    ]
+)
+@given(
+    collection=collection_st,
+    recordset=recordset_st,
+    filters=st.lists(strategies.filters(collection_st, recordset_st), min_size=1),
+)
+def test_filterable_metadata_get(caplog, api: API, collection, recordset, filters):
     caplog.set_level(logging.ERROR)
 
     api.reset()
-    coll = api.create_collection(name=collection.name,
-                                 metadata=collection.metadata,
-                                 embedding_function=collection.embedding_function)
+    coll = api.create_collection(
+        name=collection.name,
+        metadata=collection.metadata,
+        embedding_function=collection.embedding_function,
+    )
     coll.add(**recordset)
 
     for filter in filters:
         result_ids = coll.get(**filter)["ids"]
         expected_ids = _filter_embedding_set(recordset, filter)
         assert sorted(result_ids) == sorted(expected_ids)
 
+
+@settings(
+    suppress_health_check=[
+        HealthCheck.function_scoped_fixture,
+        HealthCheck.large_base_example,
+    ]
+)
+@given(
+    collection=collection_st,
+    recordset=recordset_st,
+    filters=st.lists(
+        strategies.filters(collection_st, recordset_st, include_all_ids=True),
+        min_size=1,
+    ),
+)
+def test_filterable_metadata_query(
+    caplog,
+    api: API,
+    collection: strategies.Collection,
+    recordset: strategies.RecordSet,
+    filters,
+):
+    caplog.set_level(logging.ERROR)
+
+    api.reset()
+    coll = api.create_collection(
+        name=collection.name,
+        metadata=collection.metadata,
+        embedding_function=collection.embedding_function,
+    )
+    coll.add(**recordset)
+
+    total_count = len(recordset["ids"])
+    # Pick a random vector
+    embeddings = recordset["embeddings"]
+    assert embeddings is not None
+    random_embedding = embeddings[random.randint(0, total_count - 1)]
```

Comment:
Generally, using non-Hypothesis randomness in tests is discouraged (even though it does have some hooks to preserve reproducibility).

For the purposes of this test, is there any reason not just to use the embedding at index 0? We don't actually care about the distances of the results.

Reply:
My understanding was that hypothesis sets the numpy RNG seed?

"""The Hypothesis engine handles PRNG state for the stdlib and Numpy random
    modules internally, always seeding them to zero and restoring the previous
    state after the test."""
    
    https://hypothesis.readthedocs.io/en/latest/data.html#hypothesis.strategies.random_module